python/paddle/distributed/auto_parallel/operators/common.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+DISTRIBUTED_OPERATORS = {}
+
+
+class DistributedOperator:
+    def __init__(self):
+        self._impls = []
+        self._name = None
+
+    def register_impl(self, dist_impl):
+        self._impls.append(dist_impl)
+
+    def get_impl(self, impl_idx):
+        return self._impls[impl_idx]
+
+    def get_impls(self):
+        return self._impls
+
+
+class DistributedOperatorImpl:
+    def __init__(self):
+        self._name = None
+
+    def forward(self, serial_op):
```

Comment:
serial_op object as the only one input might not be sufficient?
at least need another input argument to pass  context information (like the varname mapping in the program or graph)

Reply:
Done!